### PR TITLE
CSP-764 Implements GET CalendarEvent by ID

### PR DESCRIPTION
**Pull request by SawNawDfE**

### CODE REVIEW

**Review by DevenKShah — DISMISSED** [review]

**Review by DevenKShah — COMMENTED** [review]
We could have just included calendar navigation property to reach calendar name instead of a completely separate repository.

**Comment by SawNawDfE** [comment]
> We could have just included calendar navigation property to reach calendar name instead of a completely separate repository.

Done @DevenKShah please review

File: src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace SFA.DAS.AANHub.Data.Configuration;
+
+public class CalendarEventConfiguration : IEntityTypeConfiguration<CalendarEvent>
+{
+    [ExcludeFromCodeCoverage]
+    public void Configure(EntityTypeBuilder<CalendarEvent> builder)
+    {
+        builder.ToTable("CalendarEvent");
+        builder.HasKey(x => x.Id);
+    }
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs:15]: EF works it out by convention

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs [hunk 2]
```diff
@@ -8,11 +12,27 @@ namespace SFA.DAS.AANHub.Api.Controllers;
...
-    public CalendarEventsController(ILogger<CalendarEventsController> logger)
+    public CalendarEventsController(ILogger<CalendarEventsController> logger, IMediator mediator)
     {
         _logger = logger;
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("{calendarEventId}")]
+    [ProducesResponseType(typeof(GetMemberResult), StatusCodes.Status200OK)]
```
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:27]: Hold up, I don't think this is correct, will look into it

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**